src/test/java/org/takes/facets/auth/PsBasicDefaultTest.java

```diff
@@ -0,0 +1,122 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.facets.auth;
+
+import java.util.Arrays;
+import java.util.List;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link org.takes.facets.auth.PsBasic.Default}.
+ * @author Georgy Vlasov (wlasowegor@gmail.com)
+ * @version $Id$
+ * @since 0.22
+ */
+public final class PsBasicDefaultTest {
+    /**
+     * URN of one of users.
+     */
+    private static final String URN = "urn:foo:robert";
+    /**
+     * Login of one of users.
+     */
+    private static final String BOB_LOGIN = "bob";
+
+    /**
+     * Password of one of users.
+     */
+    private static final String BOB_PASSWORD = "qwerty";
+
+    /**
+     * Existing users.
+     */
+    private transient List<User.Default> users;
+
+    /**
+     * Sets up a list of users used in each test.
+     */
+    @Before
+    public void setUpUsersList() {
+        this.users = Arrays.asList(
+            new User.Default(
+                PsBasicDefaultTest.BOB_LOGIN,
+                PsBasicDefaultTest.BOB_PASSWORD,
+                new Identity.Simple(URN)
+            ),
+            new User.Default(
+                "alice",
+                "123",
+                new Identity.Simple("urn:foo:alice")
+            )
+        );
+    }
+    /**
+     * {@link org.takes.facets.auth.PsBasic.Default} accepts correct
+     * login/password pair.
+     */
+    @Test
+    public void defaultPassCanAcceptCorrectLoginPasswordPair() {
+        MatcherAssert.assertThat(
+            new PsBasic.Default(this.users)
+                .enter(
+                    PsBasicDefaultTest.BOB_LOGIN,
+                    PsBasicDefaultTest.BOB_PASSWORD
+            )
+                .get()
+                .urn(),
+            Matchers.equalTo(URN)
+        );
+    }
+
+    /**
+     * {@link org.takes.facets.auth.PsBasic.Default} rejects incorrect
+     * passowrd even with an existing login.
+     */
+    @Test
+    public void defaultPassCanRejectIncorrectPassword() {
+        MatcherAssert.assertThat(
+            new PsBasic.Default(this.users)
+                .enter(PsBasicDefaultTest.BOB_LOGIN, "wrongpassword")
+                .has(),
+            Matchers.is(false)
+        );
+    }
+
+    /**
+     * {@link org.takes.facets.auth.PsBasic.Default} rejects non-existing
+     * login.
+     */
+    @Test
+    public void defaultPassCanRejectIncorrectLogin() {
```

Comment:
@Suseika please see http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names